<?xml version="1.0"?>
<robot xmlns:xacro="http://www.ros.org/wiki/xacro">

  <!-- ROS2 Control setup -->
  <ros2_control name="HardwareControl" type="system">
    <hardware>
      <xacro:if value="$(arg use_sim_time)">
        <plugin>gazebo_ros2_control/GazeboSystem</plugin>
      </xacro:if>
      <xacro:unless value="$(arg use_sim_time)">
        <plugin>ipa_ros2_control/CurtMiniHardwareInterface</plugin>
      </xacro:unless>
    </hardware>
    <joint name="front_left_motor">
      <command_interface name="velocity"/>
      <state_interface name="position"/>
      <state_interface name="velocity"/>
    </joint>
    <joint name="front_right_motor">
      <command_interface name="velocity"/>
      <state_interface name="position"/>
      <state_interface name="velocity"/>
    </joint>
    <joint name="back_left_motor">
      <command_interface name="velocity"/>
      <state_interface name="position"/>
      <state_interface name="velocity"/>
    </joint>
    <joint name="back_right_motor">
      <command_interface name="velocity"/>
      <state_interface name="position"/>
      <state_interface name="velocity"/>
    </joint>
  </ros2_control>
</robot>